simulation/mdp.py

```diff
@@ -61,17 +73,59 @@ def update_sim_from_action(sim, action, debug=False):
   if debug:
     print('Action Taken:', action)
 
+"""
+Follows a predetermined sequence of controls, instead of using autopilot.
+"""
+def enact_predetermined_controls(sim, autopilot):
+  durations = [400, 200, 100, 300, 900, 400, 400,
+                  1500, 400, 1500, 400]
+  controls = [[0.8, 0., 0., 0.],
+          [0.7, 0., 0., 0.],
+          [0.5, 0., -0.2, 0.],
+          [0.4, 0., -0.13, 0.],
+          [0.29, 0., -0.01, 0.],
+          [0.29, 0., -0.01, 0.],
+          [0.29, 0., -0.01, 0.],
+          [0.22, 0.01, -0.01, 0.3],
+          [0.22, -0.01, -0.005, 0.0],
+          [0.22, 0.01, -0.01, 0.3],
+          [0.22, -0.01, -0.005, 0.0]]
+  durations = np.cumsum(durations)
+  action = [0.22, 0.0, 0.0, 0.0]
+
+  for i, duration in enumerate(durations):
+    if t < duration:
+      action = controls[i]
+      break
+  t += 1
+  if t >= times[-1]: 
+    t = 0
+  action = torch.tensor(action)
+  update_sim_from_action(sim, action)
+
+  sim.t = t
+
+  return state, action, log_prob
+  
+
 """
 Enacts the [autopilot] on the current state of the simulation [sim].
 Basically just updates sim throttle / control surfaces according to the autopilot.
 """
 def enact_autopilot(sim, autopilot):
-  state = state_from_sim(sim)
+  state = state_from_sim(sim, debug=False)
   action, log_prob = autopilot.get_controls(state)
-  update_sim_from_action(sim, action_transform(action))
+
+  update_sim_from_action(sim, action)
```

Comment:
Will need to revert this to use the network output i think

Reply:
Done. Changed to distinguish between `get_action` which is what the network outputs and `get_control` which is the direct control applied to the aircraft